addons/portal/wizard/mail_invite_share.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import werkzeug.urls
+
+from odoo import api, fields, models, _
+
+
+class MailInviteShare(models.TransientModel):
+    _name = 'mail.invite.share'
+
+    def default_share_link(self):
```

Comment:
Isn't it possible to store model and res_id on the wizard and compute a share_link based on the record ? That way we avoid relying on the context.